### ISSUE TITLE
ES dynamic template functionality is broken in Graylog 1.3

### ISSUE THREAD

**Issue by jehuty0shift** [Graylog2/graylog2-server#1624]
Hello, 

Since Graylog 1.3, the use of clever dynamic template to specify values type and doc_values is not possible anymore. Since you create the index with your own index template, you prevent anyone to specify any special mapping with a dynamic template. 
Here is the mapping we have now with graylog 1.3

```
{
  "graylog2_321" : {
    "mappings" : {
      "message" : {
        "dynamic_templates" : [ {
          "internal_fields" : {
            "mapping" : {
              "index" : "not_analyzed",
              "doc_values" : true
            },
            "match" : "gl2_*"
          }
        }, {
          "store_generic" : {
            "mapping" : {
              "index" : "not_analyzed"
            },
            "match" : "*"
          }
        },  {
          "num_suffix" : {
            "mapping" : {
              "type" : "double",
              "doc_values" : true
            },
            "match" : "*_num"
          }
        }, {
          "others" : {
            "mapping" : {
              "index" : "not_analyzed",
              "type" : "string",
              "doc_values" : true
            },
            "match" : "*"
          }
...
"properties" : {
          "thread" : {
            "type" : "long"
          },
...
```
"thread" should here a string, but it is now a long.  Note that also doc_values can not be specified anymore for non analyzed fields.
This is a problem since it was useful to prevent collisions to happen by enforcing conventions with ES.
I have also seen the dynamic_templates features as a solution proposed in some issues. I changed the order of the template but it doesn't help. It is working perfectly fine in 1.2 since the dynamic template mapping of Graylog is applied after the dynamic template present in ES.

**Comment by joschi** [comment]
Refs #1502.

**Comment by bernd** [comment]
@jehuty0shift Thank you for the report. We are able to reproduce the issue and are working on a fix.

**Comment by joschi** [comment]
Fixed in #1628.
